core/src/main/java/hudson/model/BooleanParameterDefinition.java

```diff
@@ -72,7 +73,7 @@ public BooleanParameterValue getDefaultParameterValue() {
         return new BooleanParameterValue(getName(), defaultValue, getDescription());
     }
 
-    @Extension
+    @Extension @Symbol("booleanParam")
```

Comment:
See above for my comment about why it has `param` suffix; short answer, job DSL.

The `@Symbol` annotation javadoc requires this to be a valid java identifier, which boolean is not, which is why `boolean` is not here. Do you think that constraint should be relaxed?